test/invariant/fuzz/handlers/HandlerAllo.t.sol

```diff
@@ -33,65 +34,299 @@ contract HandlerAllo is Setup {
+    function handler_updateTrustedForwarder(address _newForwarder) public {
+        // Update the trusted forwarder - will revert if caller is not the owner or if the new forwarder is zero
+        targetCall(address(allo), 0, abi.encodeWithSelector(Allo.updateTrustedForwarder.selector, _newForwarder));
+    }
+
+    function handler_addPoolManagers(uint256 _idSeed, address[] calldata _managers)
+        public
+        onlyValidPoolAndAdmin(_idSeed)
+    {
+        uint256 _poolId = _getPoolId(_idSeed);
+
+        // Add pool managers - will revert if caller is not the pool admin of the pool id
+        (bool _succ,) =
+            targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.addPoolManagers.selector, _poolId, _managers));
+
+        if (_succ) {
+            for (uint256 _i; _i < _managers.length; ++_i) {
+                ghost_poolManagers[_poolId].push(_managers[_i]);
+            }
+        }
+    }
+
+    function handler_removePoolManagers(uint256 _idSeed) public onlyValidPoolAndAdmin(_idSeed) {
+        uint256 _poolId = _getPoolId(_idSeed);
+        address[] memory _managers = ghost_poolManagers[_poolId];
+
+        // Remove pool managers - will revert if caller is not a pool admin of the pool id
+        (bool _succ,) =
+            targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.removePoolManagers.selector, _poolId, _managers));
+
+        if (_succ) {
+            delete ghost_poolManagers[_poolId];
+        }
+    }
+
+    function handler_addPoolManagersInMultiplePools(uint256[] calldata _seeds, address[] calldata _managers)
+        public
+        onlyValidPoolsAdmin(_seeds)
+    {
+        uint256[] memory _poolIds = _getPoolsIds(_seeds);
+
+        // Add pool managers in multiple pools - will revert if caller is not a pool admin of any pool id
+        (bool _succ,) = targetCall(
+            address(allo), 0, abi.encodeWithSelector(IAllo.addPoolManagersInMultiplePools.selector, _poolIds, _managers)
+        );
+
+        if (_succ) {
+            for (uint256 _i; _i < _poolIds.length; ++_i) {
+                uint256 _poolId = _poolIds[_i];
+                for (uint256 _j; _j < _managers.length; ++_j) {
+                    ghost_poolManagers[_poolId].push(_managers[_j]);
+                }
+            }
+        }
+    }
+
+    function handler_removePoolManagersInMultiplePools(uint256[] calldata _seeds, address[] calldata _managers)
+        public
+        onlyValidPoolsAdmin(_seeds)
+    {
+        uint256[] memory _poolIds = _getPoolsIds(_seeds);
+
+        // Remove pool managers in multiple pools - will revert if caller is not a pool admin of any pool id
+        (bool _succ,) = targetCall(
             address(allo),
             0,
-            abi.encodeWithSelector(
-                IAllo.updatePoolMetadata.selector,
-                poolId,
-                metadata
-            )
+            abi.encodeWithSelector(IAllo.removePoolManagersInMultiplePools.selector, _poolIds, _managers)
```

Comment:
the _managers array should come from the ghost then no? But it looks like heavier logic, as we'd need to insure every manager is indeed part of each pools before removing it

Reply:
correct 👍